lib/mix/tasks/scenic.gen.component.ex

```diff
@@ -0,0 +1,74 @@
+defmodule Mix.Tasks.Scenic.Gen.Component do
+  @shortdoc "Creates scaffolding for a new Scenic component."
+
+  @moduledoc """
+  Generates Scene boilerplate for a new component.
+
+  ```bash
+  mix scenic.gen.component component_name
+  ```
+  """
+  use Mix.Task
+
+  import Mix.Generator
+  alias ScenicNew.Common
+
+  @switches [
+    app: :string,
+    module: :string
+  ]
+
+
+  # --------------------------------------------------------
+  @doc false
+  def run(argv) do
+    if Mix.Project.umbrella?() do
+      Mix.raise "mix scenic.gen.component must be invoked from within your scenic application root directory."
+    end
+
+    {opts, argv} = OptionParser.parse!(argv, strict: @switches)
+
+    case argv do
+      [] ->
+        Mix.Tasks.Help.run(["scenic.gen.component"])
```

Comment:
To satisfy `Elixir`'s formatter:
```suggestion
  @task "scenic.gen.component"

  # --------------------------------------------------------
  @doc false
  def run(argv) do
    if Mix.Project.umbrella?() do
      message = "mix #{@task} must be invoked from within your scenic application root directory."
      Mix.raise(message)
    end

    {opts, argv} = OptionParser.parse!(argv, strict: @switches)

    case argv do
      [] ->
        Mix.Tasks.Help.run([@task])
```